python/paddle/distributed/launch/controllers/collective.py

```diff
@@ -97,10 +97,14 @@ def build_pod(self):
                 "PADDLE_TRAINERS_NUM": "{}".format(global_size),
                 "PADDLE_RANK_IN_NODE": str(i),
             }
-            if self.pod.replicas == 1:
-                e.update({selected_dev_key: ",".join(selected_dev_list)})
+            if len(selected_dev_list) > 0:
```

Comment:
Why change this file?

Reply:
Default to cpu/gloo mode if gpu is not used.